### PR TITLE
Add strict-internal-urls flag to disable domain and subpath matching

### PR DESCRIPTION
**Pull request by hbridge**
Hello, I created this so that Google Meet links don't open in my Google Calendar app for me, but it looks like others have a similar issue (e.g. issue #1304).  Feedback very welcome –– it's been awhile ;)

### CODE REVIEW

**Comment by ronjouch** [comment]
:wave: @TheCleric internalUrls-guru, do you have some time to review this one?

**Comment by TheCleric** [comment]
> :wave: @TheCleric internalUrls-guru, do you have some time to review this one?

Will do.

File: app/src/helpers/helpers.test.ts [hunk 1]
```diff
@@ -6,50 +6,74 @@ import {
...
 const externalUrl = 'https://www.wikipedia.org/wiki/Electron';
 const wildcardRegex = /.*/;
 
 test('the original url should be internal', () => {
-  expect(linkIsInternal(internalUrl, internalUrl, undefined)).toEqual(true);
+  expect(linkIsInternal(internalUrl, internalUrl, undefined, false)).toEqual(true);
 });
 
-test('sub-paths of the original url should be internal', () => {
+test('the original url should be internal with --strict-internal-urls on', () => {
```
Comment (TheCleric) [inline app/src/helpers/helpers.test.ts:25]: I would also do a test with the value set to undefined since that is supported.

**Comment by TheCleric** [comment]
LGTM other than the one extra test I'd like to see.

**Comment by TheCleric** [comment]
@ronjouch LGTM

**Comment by ronjouch** [comment]
Thanks folks.

@hbridge one last thing: this needs doc in `API.md`, after the `internal-urls` section.

**Comment by hbridge** [comment]
Ah of course, good catch @ronjouch.  Just pushed an update with docs.  One question that came up for me when writing them is that this does NOT disable login page matching, is there a scenario in which someone will want to do that and if so should I either a) also disable login page matching or b) change the name to something more targeted like --disable-base-domain-matching (wordy, I know)

**Comment by ronjouch** [comment]
> One question that came up for me when writing them is that this does NOT disable login page matching, is there a scenario in which someone will want to do that and if so should I either a) also disable login page matching or b) change the name to something more targeted like --disable-base-domain-matching (wordy, I know)

@TheCleric any preference about that?

**Comment by TheCleric** [comment]
> > One question that came up for me when writing them is that this does NOT disable login page matching, is there a scenario in which someone will want to do that and if so should I either a) also disable login page matching or b) change the name to something more targeted like --disable-base-domain-matching (wordy, I know)
> 
> @TheCleric any preference about that?

I think the logins should still work, as that seems to not be the use case for this flag. I'd say if we wanted to disable those that would be a separate flag like `--no-internal-login-pages` and in the meantime just document that this flag does not affect login pages.

**Comment by ronjouch** [comment]
@hbridge merged! Thanks again for the patch, and the back & forth on the PR. Will ship in the next release.

**Pull request merged**